core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -342,6 +342,13 @@ public final double readDouble() throws IOException {
         return Double.longBitsToDouble(readLong());
     }
 
+    public final Double readOptionalDouble() throws IOException {
```

Comment:
IN general, when we add new methods in StreamInput and StreamOuput, shall we test them e.g. adding specific lines in BytesStreamTests? We probably should have done for other methods that we recently added too...